packages/api-fetch/src/index.js

```diff
@@ -75,7 +95,13 @@ const defaultFetchHandler = ( nextOptions ) => {
 		headers[ 'Content-Type' ] = 'application/json';
 	}
 
-	const responsePromise = window.fetch( url || path, {
+	const resolvedFetchUrl = url || path;
+
+	if ( typeof resolvedFetchUrl === 'undefined' ) {
+		throw new Error( 'Please specify either a `path` or `url`' );
+	}
+
+	const responsePromise = window.fetch( resolvedFetchUrl, {
```

Comment:
Upon further investigation it turns out that it _is_ possible to pass `undefined` to `fetch`:

<img width="540" alt="window.fetch being passed undefined and an empty object for its arguments and then logging a response from the current location" src="https://user-images.githubusercontent.com/24264157/112326692-f4baad80-8c71-11eb-9409-5820be7b5238.png">

Maybe the correct default is `window.location` rather than throwing an error, to preserve this behavior? What do folks think?

Reply:
I'd very much prefer keeping the original behavior here. If we decide to change it, I'd suggest that it is to be done separately from typing the package.

Reply:
Done, just falling back to `window.location.toString()` now.